Source/Core/VideoBackends/OGL/TextureConverter.cpp

```diff
@@ -241,12 +241,12 @@ static void EncodeToRamUsingShader(GLuint srcTexture,
 	// .. and then read back the results.
 	// TODO: make this less slow.
 
-	int writeStride = bpmem.copyMipMapStrideChannels * 32;
+	//int writeStride = bpmem.copyMipMapStrideChannels * 32;
 	int dstSize = dstWidth*dstHeight*4;
-	int readHeight = readStride / dstWidth / 4; // 4 bytes per pixel
-	int readLoops = dstHeight / readHeight;
+	//int readHeight = readStride / dstWidth / 4; // 4 bytes per pixel
+	//int readLoops = dstHeight / readHeight;
 
-	if (writeStride != readStride && readLoops > 1)
+	if (dstStride != dstWidth)
```

Comment:
For xfb: Did you miss the factor bytes_per_pixel? I guess dstStride is in bytes and dstWidth is in pixels? If not, please comment that.